compiler/hash-ast/src/tree.rs

```diff
@@ -0,0 +1,584 @@
+use hash_utils::tree_writing::TreeNode;
```

Comment:
Wow holy shit, yeah we will need to look into having a macro solution for all of this code.

Reply:
Yes, although that would be for visitor.rs, not tree.rs.